tf_trees/README.md

```diff
@@ -1,18 +1,47 @@
 # The Tree Ensemble Layer
 
-This repository contains a new *tree ensemble layer* which can be used anywhere in a neural network. We provide a low-level Tensorflow implementation along with a high-level Keras API.
+This repository contains a new *tree ensemble layer* (TEL) for neural networks. The layer is differentiable so SGD can be used to train the neural network (including TEL). The layer supports conditional computation for both training and inference, i.e., when updating/evaluating a certain node in the tree, only the samples that reach that node are used in computations (this is to be contrasted with the dense computations in neural networks). We provide a low-level Tensorflow implementation along with a high-level Keras API.
 
-The layer is differentiable so SGD can be used to train the neural network (including the tree layer). The layer supports conditional computation for training and inference: when updating/evaluating a certain node in the tree, only the samples that reach that node are used in computations (this is to be contrasted with the dense computations in neural networks).
+More details to be added soon.
+
+## Installation
+The installation instructions below assume that you have Python, TensorFlow, and GCC already installed.
+
+From inside the directory containing TEL's code, run the following commands:
+```bash
+TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
+TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
+g++ -std=c++11 -shared neural_trees_ops.cc neural_trees_kernels.cc neural_trees_helpers.cc -o neural_trees_ops.so -fPIC ${TF_CFLAGS[@]} ${TF_LFLAGS[@]} -O2
+```
+Note: On OS X, add the flag "-undefined dynamic_lookup" (without quotes) to the last command above.
 
 ## Example Usage
 In Keras, the layer can be used as follows:
 ```python
-from keras.models import Sequential
-from neural_trees_layer import NeuralTrees
+import tensorflow as tf
+from tensorflow import keras
+
+# Import the layer.
+from tel import TEL
+# The documentation of TEL can be accessed as follows
+?TEL
 
-model = Sequential()
-# Add you favorite layers here...
-tree_layer = NeuralTrees(output_logits_dim=10, trees_num=5, depth=2)
+# We will fit TEL on the Boston Housing regression dataset.
```

Comment:
This seems like a full blown example, can we move it out to a sep .py file. Here you can leave
model = keras.Sequential()
model.add(keras.layers.BatchNormalization())
model.add(tree_layer)
model.add(other_layers...)
etc
or something like this